File: lxrbckl/gpt.py
# import <
from openai import AsyncOpenAI

# >


class gpt:
   
   def __init__(self, token):
      '''  '''
      
      self.openai = AsyncOpenAI(api_key = token)

      
   async def message(
      
      self,
      message,
      
      role = 'user',
      model = 'gpt-3.5-turbo'
      
   ):
      '''  '''
      
      chat = await self.openai.chat.completions.create(
         
         model = model,
         messages = [{
            
            "role" : role,
            "content" : message
            
         }]
         
      )
      
      return chat.choices[0].message.content

File: source/fp/gpt.js
// import <
const {OpenAI} = require('openai');

// >


class gpt {

   constructor({
      
      token,
      role = 'user',
      model = 'gpt-3.5-turbo'
   
   }) {

      this.role = role;
      this.model = model;
      this.openai = new OpenAI({apiKey : token});

   }


   async message({query}) {

      return (await this.openai.chat.completions.create({

         model : this.model,
         messages : [{

            content : query,
            role : this.role

         }]

      })).choices[0].message.content;

   }

}


// export <
module.exports = gpt;

// >